src/Json/SchemaLoader.php

```diff
@@ -54,14 +56,30 @@ public function __construct(FileLocatorInterface $fileLocator, DereferencerInter
     /**
      * @param string $file
      *
+     * @throws FileLoaderLoadException when given file does not have a valid JSON or YAML extension
+     *
      * @return stdClass
      */
     public function load(string $file): stdClass
     {
         $locatedFile = $this->fileLocator->locate($file);
 
         if (isset($this->schemas[$locatedFile]) === false) {
-            $schema = $this->dereferencer->dereference('file://'.$locatedFile);
+            switch (pathinfo($locatedFile, PATHINFO_EXTENSION)) {
+                case 'yml':
+                case 'yaml':
+                    $dereference = Yaml::parseFile($locatedFile, Yaml::PARSE_OBJECT_FOR_MAP);
+                    break;
+
+                case 'json':
+                    $dereference = "file://{$locatedFile}";
+                    break;
+
+                default:
+                    throw new FileLoaderLoadException($locatedFile);
```

Comment:
converted to a `switch/case` block which inspects the extension to take the appropriate action or throw an exception ... is `.json5` supported?

Reply:
Looks good. 👍 PHP doesn't have native support for JSON5 I believe.